### PR TITLE
TSS-958: History V2 POC

### PR DESCRIPTION
**Pull request by abarolo**
The barrier history endpoint scales inefficiently.

Legacy query complexity: ~O(N x M  x R) 
 
N= # historical tables
M = # items
R = # item related lookups

Below is a screenshot of a legacy barrier history tab with ~20 historical items on separate tables.

Screenshot 2023-11-18 at 23 45 19

This is a POC of working functionality backwards compatible with the remaining non-refactored legacy implementation.

Once all historical components are fully refactored:

V2 query complexity = O(N)

N = # historical tables

TODO: 
- add get_history() to all tracked models.
- delete all historical factory and item tech debt ~2000 lines + ~20 files
- certain historical data items may need enrichment, so a solution to that.

### CODE REVIEW

**Review by ferozerub — APPROVED** [review]

File: tests/history/factories.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import factory
+
+from api.barriers.models import Barrier, ProgrammeFundProgressUpdate
+
+
+class BarrierFactory(factory.django.DjangoModelFactory):
```
Comment (chris-pettinga) [inline tests/history/factories.py:6]: We already have a BarrierFactory here - https://github.com/uktrade/market-access-api/blob/588198500a067b78230b492de3e4425f7f8163ca/tests/barriers/factories.py#L78

**Review by chris-pettinga — CHANGES_REQUESTED** [review]

File: tests/history/factories.py (hunk 1 above)
Comment (abarolo) [reply tests/history/factories.py:6]: Yes I got it from here and removed parts of it. Just trying to maintain as much separation for now while doing the refactor.

Comment (chris-pettinga) [reply tests/history/factories.py:6]: ok happy with this as long as you keep track of all the additional code infrastructure the refactor needs, so we're able to quickly delete all of it/all of the old stuff when we do the absolute switchover.

**Review by chris-pettinga — APPROVED** [review]

File: tests/history/factories.py (hunk 1 above)
Comment (abarolo) [reply tests/history/factories.py:6]: of coourse

**Pull request merged**